lib/_http_client.js

```diff
@@ -30,11 +30,18 @@ function ClientRequest(options, cb) {
 
   var agent = options.agent;
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
+
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
-    agent = defaultAgent;
+  } else if (agent == null) {
```

Comment:
I misread the code, I see now, you pulled the check on createConnection out of the conditional, so that if it is a function, no check is made on the type of agent. Code like this is a bit hard to read, I suggest just adding a comment between line 39 and 40 below, making it clear that this case is explicitly being dropped through - that agent can be `== null` if there is a createConnection function, that would have caused me to not misread.